simulator/simulator.cpp

```diff
@@ -90,7 +99,9 @@ class SimulatorFactory {
             throw InvalidISA( name);
         }
 
-        return it->second.get();
+        auto tbuilder = it->second.get();
+        tbuilder->isa = name;
+        return tbuilder;
```

Comment:
I suggest to add string parameter to TBuilder ctor and create the following method:
```c++
template<typename T, Endian e>
static void emplace( Map* map, std::string_view name)
{
    map->emplace( name, std::make_unique<TBuilder<T, e>>( name)));
]
```

so it may be used instead of `my_map->emplace("name", std::<<><><>>)` boilerplate.